llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -2405,13 +2405,22 @@ bool SIInstrInfo::expandPostRAPseudo(MachineInstr &MI) const {
     // the encoding of $symbol starts 12 bytes after the start of the s_add_u32
     // instruction.
 
+    int64_t Adjust = 0;
+    if (ST.hasDelayAlu()) {
+      // Manually add the 1 cycle delay before using RegLo. AMDGPUInsertDelayAlu
+      // will not add this automatically inside a bundle:
+      //   s_delay_alu instid0(SALU_CYCLE_1)
+      Bundler.append(BuildMI(MF, DL, get(AMDGPU::S_DELAY_ALU)).addImm(9));
```

Comment:
Open question: should the existing command line flag `-amdgpu-enable-delay-alu=0` disable this? Or should that flag only disable running the `AMDGPUInsertDelayAlu` pass?

Reply:
The main use seems to be to minimize the diff from gfx10, so probably

Reply:
Done. Though now I'm wondering if this should also be disabled by `-O0`, like `AMDGPUInsertDelayAlu` is.